docs/src/main/asciidoc/extension-maturity-model.adoc

```diff
@@ -0,0 +1,88 @@
+The order in this model isn't exact. Different developers will have different views on what capabilities are most important. You may wish to (say) prioritise performance over enhancing your extension's Dev UI tile. That's fine!
+Also, not every step will apply to every extension. For example, you don't need a Dev Service if your extension doesn't depend on external services.
+It's completely OK to publish a first version of an extension that doesn't handle everything. In fact, it's OK if your extension _never_ gets to the more advanced features. This is a suggested pathway, not a minimum feature set.
+
+Also note that this list only includes the technical features of your extension.
+You might also want to think about how you share your extension, and how it presents itself to the world.
+The link:https://hub.quarkiverse.io/checklistfornewprojects/[new extension checklist] on the Quarkiverse Hub has a useful list of ways extensions can participate in the ecosystem.
+It's also a good idea to spend some time on the metadata in the link:https://quarkus.io/version/main/guides/extension-metadata#quarkus-extension-yaml[`quarkus-extension.yaml` file], which is used by Quarkus tooling.
+
+Here are some pointers on how to achieve those capabilities.
+
+== Runtime actually works
+
+=== Works in JVM mode
+
+For most extensions, this is the minimum expectation.
+When wrapping an existing library, this is usually trivial to achieve; if an extension is providing net-new capability, it might be a bit more work. Quarkus provides tools for xref:writing-extensions.adoc#testing-extensions[unit testing and integration testing] extensions.
+
+=== Works in dev mode
+
+In some cases, extra work may be needed to ensure any wrapped libraries can tolerate
+dev mode, since the classloading is different and hot reloading can break some assumptions. Extensions may also wish to add some
+xref:writing-extensions.adoc#integrating-with-development-mode[special handling for dev mode].
+To add automated tests which validate dev mode, you can xref:writing-extensions.adoc#testing-hot-reload[add tests which extend the `QuarkusDevModeTest`].
+
+=== Works as a native application
+
+For many libraries, native mode support is the primary motivation for creating an extension. See xref:writing-extensions.adoc#native-executable-support[the guide on native executable support] for more discussion about some of the adaptations that might be needed.
+
+== Configuration support
+
+Extensions should support Quarkus's unified configuration, by xref:writing-extensions.adoc#configuration[integrating with the Quarkus configuration model].
+The Writing Extensions guide has more guidance on xref:writing-extensions.adoc#how-to-expose-configuration[the Quarkus configuration philosophy].
+
+== Dev service (if there is an external service)
+
+To provide a Dev Service, use the `DevServicesResultBuildItem` build item. See the xref:extension-writing-dev-service.adoc[Dev Services how-to] for more information.
```

Comment:
```suggestion
== Dev service

Dev services are generally relevant for extensions that "connects" to something. i.e. databases for datasources, keycloak instance for security, Apache Kafka instance for kafka, etc.

If relevant for your extension, it can provide a Dev Service, use the `DevServicesResultBuildItem` build item. See the xref:extension-writing-dev-service.adoc[Dev Services how-to] for more information.
```

Just a suggestion as I think many will miss the "(if there is an external service)" in the header.

Reply:
Let's do both, and keep the header plus add the text.